vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -99,17 +99,43 @@
      */
     // <editor-fold desc="Configuration">
 
+    private static class LazyInitializationProvider {
+
+        @NotNull
+        private final Configuration mConfiguration;
+
+        public LazyInitializationProvider(@NotNull Configuration configuration) {
+            mConfiguration = configuration;
+        }
+
+        @NotNull
+        public Configuration getConfiguration() {
+            return mConfiguration;
+        }
+    }
+
+    private static LazyInitializationProvider sLazyInitializationProvider;
     private static VimeoClient mSharedInstance;
 
     public static VimeoClient getInstance() {
```

Comment:
We should probably synchronize this `getInstance` method so that it can safely be called from multiple threads